src/System.Private.CoreLib/shared/System/Globalization/CalendarData.Unix.cs

```diff
@@ -134,16 +134,103 @@ private static bool EnumDatePatterns(string localeName, CalendarId calendarId, C
             {
                 List<string> datePatternsList = callbackContext.Results;
 
-                datePatterns = new string[datePatternsList.Count];
                 for (int i = 0; i < datePatternsList.Count; i++)
                 {
-                    datePatterns[i] = NormalizeDatePattern(datePatternsList[i]);
+                    datePatternsList[i] = NormalizeDatePattern(datePatternsList[i]);
                 }
+
+                if (dataType == CalendarDataType.ShortDates)
+                    FixDefaultShortDatePattern(datePatternsList);
+
+                datePatterns = datePatternsList.ToArray();
             }
 
             return result;
         }
 
+        // FixDefaultShortDatePattern will convert the default short date pattern from using 'yy' to using 'yyyy'
+        // And will ensure the original pattern still exist in the list
+        private static void FixDefaultShortDatePattern(List<string> shortDatePatterns)
```

Comment:
I might be missing context but are we assuming there will be only one 'yy'?

Reply:
Yes. if we have more than one yy that will be ok because we'll convert at least one of them which will show the 4 digit year